.github/actions/context/action.yaml

```diff
@@ -93,21 +87,33 @@ runs:
         echo ORIGINAL_IMAGE=${ORIGINAL_IMAGE} >> ${GITHUB_OUTPUT}
         echo BUILD_REGISTRY=${BUILD_REGISTRY} >> ${GITHUB_OUTPUT}
         echo BUILD_REPO=${BUILD_REPO} >> ${GITHUB_OUTPUT}
-        echo BUILD_TAG=${BUILD_TAG} >> ${GITHUB_OUTPUT}
-        echo BUILD_IMAGE=${BUILD_IMAGE} >> ${GITHUB_OUTPUT}
       shell: bash
     - name: Generate metadata
       id: meta
       uses: docker/metadata-action@v4
       with:
         images: ${{ steps.get_context.outputs.BUILD_REGISTRY }}/${{ steps.get_context.outputs.BUILD_REPO }}
         flavor: |
-          latest=false
+          latest=true
+        tags: |
+          type=schedule
+          type=ref,event=branch
+          type=ref,event=tag
+          type=ref,event=pr
+          type=sha
     - name: Show context
       id: show_context
       run: |
-        BRANCH_TAG=$(echo "${{ steps.meta.outputs.tags }}" | cut -d ':' -f 2)
-        echo BRANCH_TAG=${BRANCH_TAG} >> ${GITHUB_OUTPUT}
+        PREFIX=$(echo "${{ steps.get_context.outputs.BUILD_REGISTRY }}/${{ steps.get_context.outputs.BUILD_REPO }}:" | sed 's%/%\/%g')
+        TAGS="${{ steps.meta.outputs.tags }}"
+        REF_TAGS="${TAGS//${PREFIX}/}"
+        BUILD_IMAGE=$(echo "${TAGS}" | tail -2 | head -1)
+        BUILD_TAG="${BUILD_IMAGE//${PREFIX}/}"
+        [[ ${BUILD_TAG} == "sha-"* ]] || exit 1 # check as parsing of the BUILD_TAG maybe fragile and dependent on docker/metadata-action priorities
```

Comment:
This breaks the pipeline right? How do we recover should the parsing be fragile ...?

Reply:
I certainly do hope it is not. If that were to happen, I would above distinctly grep the one that is tagged with `sha-...` which could be done either way. Currently, sort of trust it to be the second to last which is supported by the metadata action prioritization. As they mention it explicitly, I guess they are aware that it should not simply be re-ordered